lib/smb/ntlmssp/ntlmssp.go

```diff
@@ -172,16 +172,21 @@ func (s *AvPairSlice) UnmarshalBinary(buf []byte, meta *encoder.Metadata) error
 	if !ok {
 		return errors.New(fmt.Sprintf("Cannot unmarshal field '%s'. Missing offset\n", meta.CurrField))
 	}
-	for i := l; i > 0; {
+	offset := int64(o)
+	length := int64(l)
+	if offset+length > int64(len(meta.ParentBuf)) {
```

Comment:
Since [encoder.Metadata.Lens / Offsets are uint64](https://github.com/zmap/zgrab2/blob/master/lib/smb/smb/encoder/encoder.go#L20-L21), does this need to check that `offset` and `length` are individually still positive after the typecast, too?

Reply:
good catch.  yes, I think that should be checked too